app/script/notification/NotificationRepository.js

```diff
@@ -138,9 +138,18 @@ z.notification.NotificationRepository = class NotificationRepository {
   notify(messageEntity, connectionEntity, conversationEntity) {
     return Promise.resolve().then(() => {
       const isEventToNotify = NotificationRepository.EVENTS_TO_NOTIFY.includes(messageEntity.super_type);
-      const isMuted = conversationEntity && conversationEntity.is_muted();
-
-      const shouldNotify = isEventToNotify && !messageEntity.isEdited() && !messageEntity.isLinkPreview() && !isMuted;
+      const ignoreAllNotifications = conversationEntity && conversationEntity.showNotificationsNothing();
+
+      const showNotificationsOnlyMentions = conversationEntity && conversationEntity.showNotificationsOnlyMentions();
```

Comment:
For that kind of heavy logic code, I would probably create a new class that handles filtering the incoming notifications.
The API could be very simple (for the moment at least) with just a pure function that given a message, a connection, a conversation and a user and a notification level returns true if the notification should fire.

It will be easier in the future to bootstrap tests for that class without having to initiate all the dependencies of the NotificationRepository.

Could be done in a follow up PR